File: src/python/maps/map21.py
import ika

def AutoExec(engineRef):
    engineRef.background = engineRef.getImage('gfx/mountains.png')

def to19(engineRef):
    yield from engineRef.mapSwitchTask('map19.ika-map', (engineRef.player.sprite.x, 1 * 16))

def to22(engineRef):
    yield from engineRef.mapSwitchTask('map22.ika-map', (1 * 16, engineRef.player.sprite.y))

def to23(engineRef):
    yield from engineRef.mapSwitchTask('map23.ika-map', (48 * 16, engineRef.player.sprite.y))

def lowerPath(engineRef):
    engineRef.player.sprite.layer = 1
    if False:
        yield None

def middlePath(engineRef):
    engineRef.player.sprite.layer = 3
    if False:
        yield None

def upperPath(engineRef):
    engineRef.player.sprite.layer = 4
    if False:
        yield None

File: src/winter/maps/map21.ts
import { MapScript } from "./mapscript.js"
import { PyEngine } from "./../winter.js"

export default new MapScript(autoexec, { to19, to22, to23, lowerPath, middlePath, upperPath })

export function autoexec(engineRef: PyEngine) {
    const engine = engineRef.getEngine()
    engineRef.pySetBackground(engine.getImage('gfx/mountains.png'))
}

export function *to19(engineRef: PyEngine) {
    const p = engineRef.getPlayerEntity()
    yield* engineRef.mapSwitchTask('map19.ika-map', [p.sprite.x, 1 * 16])
}

export function *to22(engineRef: PyEngine) {
    const p = engineRef.getPlayerEntity()
    yield* engineRef.mapSwitchTask('map22.ika-map', [1 * 16, p.sprite.y])
}

export function *to23(engineRef: PyEngine) {
    const p = engineRef.getPlayerEntity()
    yield* engineRef.mapSwitchTask('map23.ika-map', [48 * 16, p.sprite.y])
}

export function *lowerPath(engineRef: PyEngine): IterableIterator<null> {
    const p = engineRef.getPlayerEntity()
    p.sprite.layer = 1
}

export function *middlePath(engineRef: PyEngine): IterableIterator<null> {
    const p = engineRef.getPlayerEntity()
    p.sprite.layer = 3
}

export function *upperPath(engineRef: PyEngine): IterableIterator<null> {
    const p = engineRef.getPlayerEntity()
    p.sprite.layer = 4
}
